src/core_plugins/tile_map/public/coordinate_maps_visualization.js

```diff
@@ -19,76 +23,91 @@ export function CoordinateMapsVisualizationProvider(Notifier, Private) {
+        previousPrecision = this._kibanaMap.getGeohashPrecision();
+        const geohashAgg = this._getGeoHashAgg();
+        const isAutoPrecision = typeof geohashAgg.params.autoPrecision === 'boolean' ? geohashAgg.params.autoPrecision : true;
+        if (geohashAgg && isAutoPrecision) {
+          geohashAgg.params.precision = previousPrecision;
         }
       });
       this._kibanaMap.on('zoomend', () => {
-        const isAutoPrecision = _.get(this._chartData, 'geohashGridAgg.params.autoPrecision', true);
+        const agg = this._getGeoHashAgg();
+        const isAutoPrecision = typeof agg.params.autoPrecision === 'boolean' ? agg.params.autoPrecision : true;
         if (!isAutoPrecision) {
           return;
         }
         if (precisionChange) {
           this.vis.updateState();
         } else {
-          this._updateData(this._chartData);
+          //when we filter queries by collar
+          this._updateData(this._rawTabifiedResponse);
         }
       });
 
       this._kibanaMap.addDrawControl();
       this._kibanaMap.on('drawCreated:rectangle', event => {
-        this.addSpatialFilter(_.get(this._chartData, 'geohashGridAgg'), 'geo_bounding_box', event.bounds);
+        const geoAgg = this._getGeoHashAgg();
+        this.addSpatialFilter(geoAgg, 'geo_bounding_box', event.bounds);
       });
       this._kibanaMap.on('drawCreated:polygon', event => {
-        this.addSpatialFilter(_.get(this._chartData, 'geohashGridAgg'), 'geo_polygon', { points: event.points });
+        const geoAgg = this._getGeoHashAgg();
+        this.addSpatialFilter(geoAgg, 'geo_polygon', { points: event.points });
       });
     }
 
     async _updateData(esResponse) {
       // Only recreate geohash layer when there is new aggregation data
       // Exception is Heatmap: which needs to be redrawn every zoom level because the clustering is based on meters per pixel
-      if (this._getMapsParams().mapType !== 'Heatmap' && esResponse === this._chartData) {
+      if (
+        this._getMapsParams().mapType !== 'Heatmap' &&
+        esResponse === this._rawTabifiedResponse) {
         return;
       }
 
-      this._chartData = esResponse;
-
+      //only now should we apply any transformations to the data.
       if (this._geohashLayer) {
         this._kibanaMap.removeLayer(this._geohashLayer);
         this._geohashLayer = null;
       }
-      if (!this._chartData || !this._chartData.geoJson) {
+
+      if (!esResponse) {
+        this._geoJsonFeatureCollectionAndMeta = null;
+        this._rawTabifiedResponse = null;
+        this._kibanaMap.removeLayer(this._geohashLayer);
+        this._geohashLayer = null;
         return;
       }
 
-      this._recreateGeohashLayer(this._chartData.geoJson);
+      this._rawTabifiedResponse = esResponse;
+
+      const geohashAgg = this._getGeoHashAgg();
+      this._geoJsonFeatureCollectionAndMeta = convertToGeoJson(esResponse, geohashAgg);
```

Comment:
do we need both, the tabified and geojson data ? if not i think it might make sense to move this to the response handler.